packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Marker interface for all platform specific plugin config impls.
+abstract class PlatformPlugin {
+  const PlatformPlugin();
+
+  Map<String, dynamic> toMap();
+}
+
+/// Contains all the required parameters to template an Android plugin.
+class AndroidPlugin extends PlatformPlugin {
+  const AndroidPlugin({
+    this.name,
+    this.package,
+    this.pluginClass,
+  });
+
+  factory AndroidPlugin.fromYaml(String name, dynamic yaml) {
+    assert(yaml != null);
+    return AndroidPlugin(
+      name: name,
+      package: yaml['package'],
+      pluginClass: yaml['pluginClass'],
+    );
+  }
+
+  @override
+  Map<String, dynamic> toMap() {
+    return <String, dynamic>{
+      'name': name,
+      'package': package,
+      'class': pluginClass,
+    };
+  }
+
+  static const String kConfigKey = 'android';
+
+  final String name;
+  final String package;
+  final String pluginClass;
+}
+
+/// Contains all the required parameters to template an iOS plugin.
+class IOSPlugin extends PlatformPlugin {
+  const IOSPlugin({
+    this.name,
+    this.classPrefix,
+    this.pluginClass,
+  });
+
+  factory IOSPlugin.fromYaml(String name, dynamic yaml) {
+    assert(yaml != null);
+    return IOSPlugin(
+      name: name,
+      classPrefix: yaml['classPrefix'],
+      pluginClass: yaml['pluginClass'],
+    );
+  }
+
+  @override
+  Map<String, dynamic> toMap() {
+    return <String, dynamic>{
+      'name': name,
+      'prefix': classPrefix,
+      'class': pluginClass,
+    };
+  }
+
+  static const String kConfigKey = 'ios';
+
+  final String name;
+  final String classPrefix;
+  final String pluginClass;
+}
+
+/// Contains all the required parameters to template an macOS plugin.
+class MacOSPlugin extends PlatformPlugin {
+  const MacOSPlugin({
+    this.name,
+    this.classPrefix,
```

Comment:
Please remove `classPrefix` for macOS; we don't need to support it, it was just a stand-in for indicating support. We don't have the legacy concerns of iOS.

Ideally we shouldn't have it for iOS in the new format either, since it should just be part of the `pluginClass` now that that's platform-specific, but because the template, very unfortunately, was designed such that the header filename doesn't include the class prefix, I think we're stuck with it forever (standard ObjC practice is for the filename to match the full class name, including prefix, so it's nonstandard). On the plus side, with Swift being the default template now, over time that will be less common.